src/connection-manager/connections.ts

```diff
@@ -93,18 +168,80 @@ async function handleWsMessageTypes(
   }
 }
 
+/**
+ * Handle ws ledger_entry amendments messages.
+ *
+ * @param ws - A WebSocket object.
+ * @param data - The WebSocket message received from connection.
+ * @param networks - The networks of subscribed node.
+ */
+async function handleWsMessageLedgerEntryAmendments(
+  ws: WebSocket,
+  data: LedgerEntryAmendmentsResponse,
+  networks: string | undefined,
+): Promise<void> {
+  if (data.result.node.LedgerEntryType === 'Amendments') {
+    await saveAmendmentsEnabled(data.result.node.Amendments, networks)
+  }
+  if (isFlagLedgerPlusOne(data.result.ledger_index)) {
+    getEnableAmendmentLedger(ws, data.result.ledger_index)
+  }
+}
+
+/**
+ * Handle ws ledger messages to search for EnableAmendment transactions.
+ *
+ * @param ws - A WebSocket object.
+ * @param data - The WebSocket message received from connection.
+ */
+async function handleWsMessageLedgerEnableAmendments(
+  ws: WebSocket,
+  data: LedgerEnableAmendmentResponse,
+): Promise<void> {
+  data.result.ledger.transactions.forEach(async (transaction) => {
+    if (
+      transaction.TransactionType === 'EnableAmendment' &&
+      !transaction.Flags
+    ) {
+      getEnableAmendmentTx(ws, transaction.hash)
+    }
+  })
+}
+
+/**
+ * Handle ws ledger messages to process EnableAmendment transactions.
+ *
+ * @param data - The WebSocket message received from connection.
+ * @param networks - The WebSocket message received from connection.
+ */
+async function handleWsMessageTxEnableAmendments(
+  data: TxEnableAmendmentResponse,
+  networks: string | undefined,
+): Promise<void> {
+  const amendment: AmendmentEnabled = {
+    amendment_id: data.result.Amendment,
+    networks,
+    ledger_index: data.result.ledger_index,
+    tx_hash: data.result.hash,
+    date: new Date(rippleTimeToUnixTime(data.result.date)),
+  }
+  await saveAmendmentEnabled(amendment)
+}
+
 /**
  * Sets the handlers for each WebSocket object.
  *
  * @param ip - The ip address of the node we are trying to reach.
  * @param ws - A WebSocket object.
  * @param networks - The networks of the node we are trying to reach where it retrieves validations.
+ * @param entry - Whether source node is an entry node for the network.
  * @returns A Promise that resolves to void once a connection has been created or timeout has occured.
  */
 async function setHandlers(
   ip: string,
   ws: WebSocket,
   networks: string | undefined,
+  entry = false,
```

Comment:
`entry` has two meanings - an element or the place of entrance. It's a bit of a confusing name here (I had to look at the docstring to understand what it meant because I thought it was referring to the response at first) - perhaps something like `isInitialNode` would be a better name.